### PR TITLE
Making the ES operator PSA compliant

### PR DESCRIPTION
**Pull request by shwetaap**
Signed-off-by: Shweta Padubidri <spadubid@redhat.com>

### Description
Making the ES operator PSA compliant

/cc @Red-GV 
/assign @periklis 

### Links

- JIRA: [LOG-2736](https://issues.redhat.com//browse/LOG-2736)

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -28,6 +28,9 @@ spec:
         resources: {}
```
Comment (aminesnow) [inline config/manager/manager.yaml:28]: No `runAsNonRoot` for the pod?

Comment (shwetaap) [reply config/manager/manager.yaml:28]: Added it here instead - https://github.com/openshift/elasticsearch-operator/pull/917/files#diff-7263fc341f1b59afdc8599e1f570cdd180c28bf8cca71ecee1e58b58131202c5R43. Thats the same pod, but a different container.  The resulting CSV created has the `runAsNonRoot` here - https://github.com/openshift/elasticsearch-operator/pull/917/files#diff-7263fc341f1b59afdc8599e1f570cdd180c28bf8cca71ecee1e58b58131202c5R43

Comment (Red-GV) [reply config/manager/manager.yaml:28]: Hmm, in that case. I would say maybe let's put that one for the pod in `manager` vs the `proxy_patch`.

Comment (shwetaap) [reply config/manager/manager.yaml:28]: Done

**Pull request merged**